internal/controller/progressive/progressive.go

```diff
@@ -375,7 +377,7 @@ func processUpgradingChild(
 			return false, 0, err
 		}
 
-		needsUpdating, err := controller.ChildNeedsUpdating(ctx, existingUpgradingChildDef, newUpgradingChildDef)
+		needsUpdating, err := rolloutNeedsUpdating(ctx, controller, rolloutObject, existingUpgradingChildDef, newUpgradingChildDef)
```

Comment:
after Failure case, we look to see if not just the primary child's definition has changed but also if any Riders have been added, modified, or deleted